### PR TITLE
SC upgrade RavenDB

### PR DESCRIPTION
**Pull request by SzymonPobiega**
* Information about the new DB maintenance port setting
 * Deprecated the max throughput setting
 * Added upgrade guide from V1 to V2

TODO:
 * [x] Check what happens when using PowerShell to upgrade from 1.X to 2.0
 * [x] Add section to the doco that explains how long it might take by hinting at the 300 GB tests we ran
 * [x] Reemphasize again with a warning that it might take considerable amount of time to migrate the storage format plus the indexes to catch up
 * [x] Add documentation to explain the multi instance migration approach as an alternative for people who don't want to go through this long migration

### CODE REVIEW

File: servicecontrol/upgrades/servicecontrol1to2.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ - 5
+ - 5
+---
+
+
+## RavenDB data files
+
+ServiceControl Version 1 uses RavenDB 2.5 while ServiceControl Version 2 uses RavenDB 3.5. Data file formats of RavenDB 2.5 and 3.5 are different and require migration process. The migration is conducted automatically by RavenDB server (embedded in the ServiceControl process) when the database engine detects that the data files are in the old format.
+
+The migration is managed by the ServiceControl Management Utility instance upgrade process and does not require manual intervention. It is recommended, however, to back up the RavenDB data folder prior to attempting the upgrade.
```
Comment (tmasternak) [inline servicecontrol/upgrades/servicecontrol1to2.md]: Should this be a NOTE or WARN?

Comment (SzymonPobiega) [reply servicecontrol/upgrades/servicecontrol1to2.md]: Added WARN

**Comment by weralabaj** [comment]
Is this a WIP? Waiting for reviews from all other people or good to merge after one approval?

**Comment by WilliamBZA** [comment]
Marked as WIP until the TF can sync properly

File: servicecontrol/creating-config-file.md [hunk 2]
```diff
@@ -199,7 +208,9 @@ Valid range for this setting is from 1 hour to 200 days.
 
 #### ServiceControl/MaximumMessageThroughputPerSecond
 
-This setting was introduced in version 1.5. The setting controls the maximum throughput of messages ServiceControl will handle per second and is necessary to avoid overloading the underlying messages database. An appropriate limit ensures that the database can cope with the number of insert operations. Otherwise the query performance will drop significantly and the message expiration process will stop working under heavy insert load. Make sure to conduct thorough performance tests on the hardware before increasing this value.
+NOTE: This setting is not applicable in Version 2 and above.
```
Comment (mikeminutillo) [inline servicecontrol/creating-config-file.md]: Is the setting ignored? Should we respect it if it is set in config?

File: servicecontrol/upgrades/servicecontrol1to2.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ - 5
+ - 5
+---
+
+
+## RavenDB data files
+
+ServiceControl Version 1 uses RavenDB 2.5 while ServiceControl Version 2 uses RavenDB 3.5. Data file formats of RavenDB 2.5 and 3.5 are different and require migration process. The migration is conducted automatically by RavenDB server (embedded in the ServiceControl process) when the database engine detects that the data files are in the old format.
+
+WARN: The migration is managed by the ServiceControl Management Utility instance upgrade process and does not require manual intervention. It is recommended, however, to back up the RavenDB data folder prior to attempting the upgrade.
```
Comment (mikeminutillo) [inline servicecontrol/upgrades/servicecontrol1to2.md]: Do we have doco on how to do that?

File: servicecontrol/upgrades/servicecontrol1to2.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+
+## Database maintenance port
+
+In Version 1 the ServiceControl running in the maintenance mode exposed the underlying RavenDB database on the same port as it exposed the API during normal operation. This approach is no longer possible with RavenDB 3.5 hence ServiceControl Version 2 requires explicit configuration of a database maintenance port which needs to be different from the API port.
+
+ServiceControl Management Utility automatically detects the need for this new configuration setting and is going to ask the user to provide it when performing the upgrade of a ServiceControl instance.
```
Comment (mikeminutillo) [inline servicecontrol/upgrades/servicecontrol1to2.md]: What happens if I upgrade using the powershell scripts?

File: servicecontrol/creating-config-file.md (hunk 2 above)
Comment (SzymonPobiega) [reply servicecontrol/creating-config-file.md]: That setting was entirely removed so we'll ignore it from now on. The purpose of it was to ensure that the expiration policy can catch up. The tests we did with 3.5 all prove that the expiration policy can cope up with the insert load I don't think we should keep it.

File: servicecontrol/upgrades/servicecontrol1to2.md (hunk 3 above)
Comment (SzymonPobiega) [reply servicecontrol/upgrades/servicecontrol1to2.md]: We do. I'll add the link.

File: servicecontrol/upgrades/servicecontrol1to2.md (hunk 4 above)
Comment (SzymonPobiega) [reply servicecontrol/upgrades/servicecontrol1to2.md]: That's a good question. I'll try it.

**Comment by WilliamBZA** [comment]
I've added details on how to perform a side-by-side upgrade, @SzymonPobiega @mikeminutillo @danielmarbach can you please take a look at it for me?

File: servicecontrol/upgrades/servicecontrol1to2.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (danielmarbach) [inline servicecontrol/upgrades/servicecontrol1to2.md]: Should this file be deleted?

Comment (WilliamBZA) [reply servicecontrol/upgrades/servicecontrol1to2.md]: Urgh, they should be combined. I'll do that. Not sure how I could have missed this

**Comment by WilliamBZA** [comment]
Removing WIP as this is ready for full review

**Pull request merged**